packages/core/ui-settings/core-ui-settings-server-internal/src/ui_settings_service.ts

```diff
@@ -109,7 +111,7 @@ export class UiSettingsService
       const isNamespaceScope = scope === 'namespace';
       const options = {
         type: (isNamespaceScope ? 'config' : 'config-global') as 'config' | 'config-global',
-        id: version,
+        id: stripVersionQualifier(version),
```

Comment:
For consistency with how we're treating qualified versions (e.g X.Y.Z-SNAPSHOT or X.Y.Z-RC1) in other parts of core, especially in SO code, we're now stripping qualifiers when providing the version to the uiSettings client.

Note that this isn't directly related to the fix, but it was trivial enough to include it there.